tools/wt_verify.py

```diff
@@ -148,11 +159,70 @@ def parse_output():
                 is_root_node = False
     return output
 
-def visualize(data, visualization_type):
-    """
-    Visualizing data
-    """
-    pass
+
+def histogram(field, chkpt, chkpt_name):
+    internal = []
+    leaf = []
+    for val in chkpt.values():
+        if field in val:
+            if "internal" in val["page_type"]:
+                internal.append(val[field])
+            elif "leaf" in val["page_type"]:
+                leaf.append(val[field])
+    fig, (ax1, ax2) = plt.subplots(2, figsize=(15, 10))
+    ax = (ax1, ax2)
+    ax[0].hist(internal, bins=100, color=PLOT_COLORS[field][0], label="internal")
+    ax[1].hist(leaf, bins=100, color=PLOT_COLORS[field][1], label="leaf")
+    ax[0].set_title(chkpt_name + " - Comparison between internal and leaf page " + field, fontsize=TITLE_SIZE)
+
+    for subplot in ax:
+        subplot.legend()
+        if field == "entries":
+            subplot.set_xlabel(field)
+        else:
+            subplot.set_xlabel(field + ' (bytes)')
+        subplot.set_ylabel('Frequency')
+
+    imgs = mpld3.fig_to_html(fig) 
+    plt.close()
+    return imgs
+
+
+def pie_chart(field, chkpt, chkpt_name):
+    sizes = [0, 0]
+    for val in chkpt.values():
+        if field in val:
+            if "internal" in val["page_type"]:
+                sizes[0] += 1
+            elif "leaf" in val["page_type"]:
+                sizes[1] += 1
+    labels = ["internal - " + str(sizes[0]), "leaf - " + str(sizes[1])]
+
+    fig, ax = plt.subplots(figsize=(10, 10))
+    plt.title(chkpt_name + " - " + field, fontsize=TITLE_SIZE)
+    ax.pie(sizes, labels=labels, autopct='%1.1f%%', colors=PLOT_COLORS[field])
+    imgs = mpld3.fig_to_html(fig)
+    plt.close()
+    return imgs
+
+
+def visualize_field(output, field):
+    imgs = ""
+    for chkpt_name, chkpt in output.items():
```

Comment:
Perhaps 'tree' instead of `chkpt`?

Reply:
But i am iterating through checkpoints?

Reply:
Yep, that makes sense too, resolving this.